aspnetcore/grpc/why-migrate-wcf-to-dotnet-grpc.md

```diff
@@ -0,0 +1,56 @@
+
+Both platforms share the principle of declaring and then implementing an interface, even though the process for declaring that interface is different. The different types of RPC calls that gRPC supports map well to the bindings available to WCF services. For more information and examples, see [Migrate a WCF solution to gRPC](https://docs.microsoft.com/dotnet/architecture/grpc-for-wcf-developers/migrate-wcf-to-grpc).
+
+## Benefits of gRPC
+
+gRPC stands above other solutions for the following reasons.
+
+### Performance
+
+gRPC uses HTTP/2.  HTTP/2 is a smaller, faster binary protocol than HTTP/1.1, which is more efficient for computers to parse. HTTP/2 also supports multiplexing requests over a single connection. Multiplexing enables multiple requests to be sent over one connection without requests blocking each other. (In HTTP/1.1, this issue is known as "head-of-line (HOL) blocking.") gRPC uses the Protobuf, an efficient binary format, to serialize messages. Protobuf messages are small in size and use less bandwidth than text-based formats. gRPC is a good solution to use for mobile devices and over slower networks where bandwidth is important.
+
+### Interoperability
+
+There are gRPC tools and libraries for all major programming languages and platforms, including .NET, Java, Python, Go, C++, Node.js, Swift, Dart, Ruby, and PHP. Thanks to the Protocol Buffers binary wire format and the efficient code generation for each platform, developers can build performant apps while still enjoying full cross-platform support.
+
+### Usability and productivity
+
+gRPC is a comprehensive RPC solution. It works consistently across multiple languages and platforms. It also provides excellent tooling, with much of the necessary boilerplate code automatically generated. Like WCF, gRPC automatically generates messages and a strongly typed client for you. Developer time is freed up to focus on business logic.
+
+### Streaming
+
+gRPC has full bidirectional streaming, which provides similar functionality to WCF's full duplex services. gRPC streaming can operate over regular internet connections, load balancers, and service meshes.
+
+### Deadline/timeouts and cancellation
+
+gRPC allows clients to specify a maximum time for an RPC to finish. If the specified deadline is exceeded, the server can cancel the operation independently of the client. Deadlines and cancellations can be propagated through further gRPC calls to help enforce resource usage limits. Clients can stop operations when a deadline is exceeded, or earlier if necessary (for example, because of a user interaction).
```

Comment:
```suggestion
gRPC allows clients to specify a maximum time for an RPC to finish. If the specified deadline is exceeded, the server can cancel the operation independently of the client. Deadlines and cancellations can be propagated through further gRPC calls to help enforce resource usage limits. Clients can stop operations when a deadline is exceeded, or earlier if necessary. For example,  clients can stop operations because of a user interaction.
```

Reply:
Per [MT](https://review.docs.microsoft.com/en-us/help/contribute/localization-mt-guidance?branch=master) Avoid parenthetical clauses.